system_firmware conversion complete

sbeaver-netgate opened the pull request:
Ready for review

sbeaver-netgate committed:
system_firmware conversion complete

Ready for review

SjonHortensius commented on usr/local/www/system_firmware.php:
```diff
@@ -202,116 +197,128 @@ function file_upload_error_message($error_code) {
+if ($savemsg)
+	print_info_box($savemsg);
+
+if ($fwinfo <> "")
```
Cleaner should have converted this to `!=`

SjonHortensius commented on usr/local/www/system_firmware.php:
```diff
@@ -102,14 +102,9 @@ function file_upload_error_message($error_code) {
-	include("fend.inc");
-	echo "</body>";
-	echo "</html>";
+	print_info_box(gettext("An upgrade is currently in progress. The firewall will reboot when the operation is complete.") . "<p><img src='/themes/{$g['theme']}/images/icons/icon_fw-update.gif' alt='update' /></p>");
```
Is this image still present in the docroot? I'd guess it's no longer necessary

sbeaver-netgate replied:
Yes it is. It is a rather nice animated image so I left it in place. We can always remove it later

> On Apr 30, 2015, at 3:41 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/system_firmware.php https://github.com/SjonHortensius/pfsense/pull/157#discussion_r29462803:
> 
> > ```
> > include("fbegin.inc");
> > ```
> > -   echo "<div>\n";
> > -   print_info_box(gettext("An upgrade is currently in progress.<p>The firewall will reboot when the operation is complete.") . "</p><p><img src='/themes/{$g['theme']}/images/icons/icon_fw-update.gif' alt='update' /></p>");
> > -   echo "</div>\n";
> > -   include("fend.inc");
> > -   echo "</body>";
> > -   echo "</html>";
> > -   print_info_box(gettext("An upgrade is currently in progress. The firewall will reboot when the operation is complete.") . "<p><img src='/themes/{$g['theme']}/images/icons/icon_fw-update.gif' alt='update' /></p>");
> >   Is this image still present in the docroot? I'd guess it's no longer necessary
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/157/files#r29462803.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

SjonHortensius commented on usr/local/www/system_firmware.php:
```diff
@@ -202,116 +197,128 @@ function file_upload_error_message($error_code) {
+
+		if (!is_subsystem_dirty('rebootreq')) {
+			if (!is_subsystem_dirty('firmware')) {
+				$enablebtn = new Form_Button(Submit, 'Enable firmware upload', null);
```
Please use proper newlines here. Also, _Submit_ should be quoted as its a string. And the 3rd parameter is null by default, so please omit it

sbeaver-netgate committed:
Detailed changes per SH

sbeaver-netgate committed:
system_firmware detailed changes

Removed unneeded placeholders
Forced firmware upgrades to be disabled on atoll page load.
